src/V3LinkLevel.cpp

```diff
@@ -50,10 +50,15 @@ void V3LinkLevel::modSortByLevel() {
     ModVec tops;  // Top level modules
     for (AstNodeModule* nodep = v3Global.rootp()->modulesp(); nodep;
          nodep = VN_CAST(nodep->nextp(), NodeModule)) {
-        if (nodep->level() <= 2) tops.push_back(nodep);
+        if (v3Global.opt.topModule() == nodep->prettyName()) {
+            tops.insert(tops.begin(), nodep);
+        } else if (nodep->level() <= 2) {
```

Comment:
```suggestion
        } else if (v3Global.opt.topModule() == "" && nodep->level() <= 2) {
```

Then can perhaps remove below check for topModule == "".

Reply:
Updated!